core/dbt/deps/registry.py

```diff
@@ -68,9 +70,28 @@ def install(self, project, renderer):
         system.make_directory(os.path.dirname(tar_path))
 
         download_url = metadata.downloads.tarball
-        system.download_with_retries(download_url, tar_path)
         deps_path = project.packages_install_path
         package_name = self.get_project_name(project, renderer)
+
+        download_untar_fn = functools.partial(
+            self.download_and_untar,
+            download_url,
+            tar_path,
+            deps_path,
+            package_name
+        )
+        connection_exception_retry(download_untar_fn, 5)
```

Comment:
It might help with testing and minimizing retry logic to separate downloading and unpacking the tarball into two distinct steps. If the error is with unpacking, redoing the download would be unnecessary.

Reply:
@nathaniel-may we do possibly need to retry the download when unpacking the tarball fails.  We have data from the last github incident where only about half of the runs failed during the incident implying that some jobs were able to run `dbt deps` successfully.  It's impossible to know why but seems worth it to retry downloading the file in case something went wrong there.

That said, pulling them apart will make it easier to test so I'll incorporate both for the purpose of testing.